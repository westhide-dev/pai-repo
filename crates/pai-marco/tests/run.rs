//! Inspire by
//! - [cargo-expand](https://github.com/dtolnay/cargo-expand)
//! - [trybuild](https://github.com/dtolnay/trybuild)
//! - [macrotest](https://github.com/eupn/macrotest)

use insta::assert_snapshot;
use pai_error::PResult;
use pai_file::SourceFile;
use quote::ToTokens;
use syn;
use syn_select;

mod source;

#[test]
fn main() -> PResult<()> {
    let syn_file = syn::parse_str(SourceFile::read("__cache__/code.rs")?.source())?;

    let code = syn::File {
        shebang: None,
        attrs: Vec::new(),
        items: syn_select::select("source", &syn_file)?,
    }
    .into_token_stream()
    .to_string();

    Ok(assert_snapshot!(code))
}
