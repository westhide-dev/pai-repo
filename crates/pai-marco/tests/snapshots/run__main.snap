---
source: crates/pai-marco/tests/run.rs
expression: code
---
pub mod unit { use pai_marco :: ts_attribute ; pub fn foo () { } }
